File: tools/bench/answers/go_haiku/s3/04_binary.go
package main

import (
	"fmt"
)

func main() {
	nums := []int{13, 64}
	for _, n := range nums {
		if n == 0 {
			fmt.Println("0")
			continue
		}
		var binary string
		x := n
		for x > 0 {
			if x%2 == 1 {
				binary = "1" + binary
			} else {
				binary = "0" + binary
			}
			x /= 2
		}
		fmt.Println(binary)
	}
}

File: tools/bench/answers/rust_haiku/s1/04_binary.rs
fn main() {
    let nums = [13, 64];

    for num in nums.iter() {
        let mut n = *num;
        let mut binary = String::new();

        if n == 0 {
            binary.push('0');
        } else {
            while n > 0 {
                if n % 2 == 1 {
                    binary.insert(0, '1');
                } else {
                    binary.insert(0, '0');
                }
                n /= 2;
            }
        }

        println!("{}", binary);
    }
}
